OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -111,15 +113,17 @@ public override bool HandleMouseInput(MouseInput mi)
 
 						if (unit != null && unit.Owner == (World.RenderPlayer ?? World.LocalPlayer))
 						{
-							var newSelection2 = SelectActorsInBox(World, worldRenderer.Viewport.TopLeft, worldRenderer.Viewport.BottomRight,
-								a => a.Owner == unit.Owner && a.Info.Name == unit.Info.Name);
+							// Select actors on the screen that have the same selection class as the already selected actor
+							var sc = unit.Trait<Selectable>().Class;
```

Comment:
It doesn't @penev92, I've checked. I have to do the check.
@Phrohdoh, I don't see a similar case below.